File: src/main/java/com/damianryan/octopus/model/Reading.java
package com.damianryan.octopus.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

import java.time.Instant;

@Data
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Reading implements Comparable<Reading> {

    double consumption;

    @JsonProperty("interval_start")
    Instant from;

    @JsonProperty("interval_end")
    Instant to;

    @Override
    public int compareTo(Reading other) {
        return from.compareTo(other.getFrom());
    }
}

File: src/main/kotlin/com/damianryan/octopus/model/Reading.kt
package com.damianryan.octopus.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant

@JvmRecord data class Reading(
    val consumption: Double = 0.0,
    @JsonProperty("interval_start") val from: Instant,
    @JsonProperty("interval_end") val to: Instant,
) : Comparable<Reading> {
    override fun compareTo(other: Reading): Int {
        return from.compareTo(other.from)
    }
}
